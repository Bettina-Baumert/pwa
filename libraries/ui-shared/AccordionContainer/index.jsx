import { Component } from 'react';
import PropTypes from 'prop-types';

/**
 * The Accordion container component.
 */
class AccordionContainer extends Component {
  static propTypes = {
    children: PropTypes.func.isRequired,
  }

  state = {
    open: false,
  }

  /**
   * @param {Object} nextProps The next component props.
   * @param {Object} nextState The next component state.
   * @returns {boolean}
   */
  shouldComponentUpdate(nextProps, nextState) {
    return this.state.open !== nextState.open;
  }

  open = () => {
    this.setState({ open: true });
  }

  close = () => {
    this.setState({ open: false });
  }

  /**
   * @returns {JSX}
   */
  render() {
    return this.props.children({
      handleClose: this.close,
      isOpen: this.state.open,
      handleOpen: this.open,
    });
  }
}

export default AccordionContainer;
