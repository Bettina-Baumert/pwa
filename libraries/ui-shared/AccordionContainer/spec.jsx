import React from 'react';
import { shallow } from 'enzyme';
import AccordionContainer from './index';

/**
 * @returns {JSX}
 */
const Child = () => <div />;

describe('<AccordionContainer />', () => {
  it('should render children with props', () => {
    const wrapper = shallow((
      <AccordionContainer>
        {props => <Child {...props} />}
      </AccordionContainer>
    ));

    const props = wrapper.find('Child').props();

    expect(wrapper).toMatchSnapshot();
    expect(props.open).toEqual(false);
    expect(typeof props.handleOpen).toEqual('function');
    expect(typeof props.handleClose).toEqual('function');
  });

  it('should update children props when state changes', () => {
    const wrapper = shallow((
      <AccordionContainer>
        {props => <Child {...props} />}
      </AccordionContainer>
    ));

    wrapper.setState({ open: true });

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('Child').props().open).toEqual(true);
  });
});
