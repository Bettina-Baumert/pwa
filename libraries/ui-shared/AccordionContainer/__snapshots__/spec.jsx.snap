// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccordionContainer /> should render children with props 1`] = `
<Child
  handleClose={[Function]}
  handleOpen={[Function]}
  isOpen={false}
/>
`;

exports[`<AccordionContainer /> should update children props when state changes 1`] = `
<Child
  handleClose={[Function]}
  handleOpen={[Function]}
  isOpen={true}
/>
`;
